src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/TestService.kt

```diff
@@ -1,19 +1,98 @@
 package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
 
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.ToJson
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.Booking
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.TestData
 import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.repository.TestRepository
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.Optional
+import java.util.UUID
 
 @Service
 class TestService(
   private val testRepository: TestRepository,
   private val prisonService: PrisonService,
 ) {
 
+  var moshi: Moshi = Moshi.Builder()
+    .addLast(KotlinJsonAdapterFactory())
+    .add(LocalDate::class.java, LocalDateAdapter().nullSafe())
+    .add(OptionalLocalDateAdapter())
+    .add(UUID::class.java, UUIDAdapter().nullSafe())
+    .build()
+
   fun getTestData(): List<TestData> {
     val prisoner = prisonService.getOffenderDetail("A1234AA")
     val testData = testRepository.findAll().map { transform(it) }.toMutableList()
-    testData.add(TestData(prisoner.offenderNo, prisoner.bookingId))
+    testData.add(TestData(prisoner.offenderNo, prisoner.bookingId.toString()))
     return testData
   }
+
+  fun jsonToBooking(json: String): Booking {
```

Comment:
This `jsonToBooking` method has been copied from `JsonTranformer` which is in the test scope